Modules/_csv.c

```diff
@@ -1213,6 +1213,19 @@ csv_writerow(WriterObj *self, PyObject *seq)
             append_ok = join_append(self, field, quoted);
             Py_DECREF(field);
         }
+        else if (PyBool_Check(field)) {
+            PyObject *str;
+            double d = PyFloat_AsDouble(field);
+            str = PyFloat_FromDouble(d);
+            str = PyObject_Str(str);
```

Comment:
Here is a memory leak. The float object referenced by str is leaked after reassigning str.